diff --git a/ament_cmake_gmock/cmake/ament_add_gmock.cmake b/ament_cmake_gmock/cmake/ament_add_gmock.cmake
index 9c80503..4a06b4b 100644
--- a/ament_cmake_gmock/cmake/ament_add_gmock.cmake
+++ b/ament_cmake_gmock/cmake/ament_add_gmock.cmake
@@ -47,8 +47,8 @@
 # @public
 #
 macro(ament_add_gmock target)
-  _ament_cmake_gmock_find_gmock()
-  if(GMOCK_FOUND)
+  find_package(GTest REQUIRED)
+  if(TARGET GTest::gmock)
     _ament_add_gmock("${target}" ${ARGN})
   endif()
 endmacro()
@@ -69,9 +69,9 @@ function(_ament_add_gmock target)
   add_executable("${target}" ${ARG_UNPARSED_ARGUMENTS})
   target_include_directories("${target}" PUBLIC "${GMOCK_INCLUDE_DIRS}")
   if(NOT ARG_SKIP_LINKING_MAIN_LIBRARIES)
-    target_link_libraries("${target}" ${GMOCK_MAIN_LIBRARIES})
+    target_link_libraries("${target}" GTest::gmock_main)
   endif()
-  target_link_libraries("${target}" ${GMOCK_LIBRARIES})
+  target_link_libraries("${target}" GTest::gmock)
 
   set(executable "$<TARGET_FILE:${target}>")
   set(result_file "${AMENT_TEST_RESULTS_DIR}/${PROJECT_NAME}/${target}.gtest.xml")
diff --git a/ament_cmake_gtest/cmake/ament_add_gtest_executable.cmake b/ament_cmake_gtest/cmake/ament_add_gtest_executable.cmake
index 6c33581..5638e14 100644
--- a/ament_cmake_gtest/cmake/ament_add_gtest_executable.cmake
+++ b/ament_cmake_gtest/cmake/ament_add_gtest_executable.cmake
@@ -32,8 +32,8 @@
 # @public
 #
 macro(ament_add_gtest_executable target)
-  _ament_cmake_gtest_find_gtest()
-  if(GTEST_FOUND)
+  find_package(GTest REQUIRED)
+  if(TARGET GTest::gtest)
     _ament_add_gtest_executable("${target}" ${ARGN})
   endif()
 endmacro()
@@ -50,9 +50,9 @@ function(_ament_add_gtest_executable target)
   add_executable("${target}" ${ARG_UNPARSED_ARGUMENTS})
   target_include_directories("${target}" BEFORE PUBLIC "${GTEST_INCLUDE_DIRS}")
   if(NOT ARG_SKIP_LINKING_MAIN_LIBRARIES)
-    target_link_libraries("${target}" ${GTEST_MAIN_LIBRARIES})
+    target_link_libraries("${target}" GTest::gtest_main)
   endif()
-  target_link_libraries("${target}" ${GTEST_LIBRARIES})
+  target_link_libraries("${target}" GTest::gtest)
   if(NOT WIN32)
     set(THREADS_PREFER_PTHREAD_FLAG ON)
     find_package(Threads REQUIRED)
diff --git a/ament_cmake_gtest/cmake/ament_find_gtest.cmake b/ament_cmake_gtest/cmake/ament_find_gtest.cmake
index 9158be1..f5ff4f5 100644
--- a/ament_cmake_gtest/cmake/ament_find_gtest.cmake
+++ b/ament_cmake_gtest/cmake/ament_find_gtest.cmake
@@ -30,5 +30,5 @@ macro(ament_find_gtest)
     message(FATAL_ERROR
       "ament_find_gtest() called with unused arguments: ${_ARGN}")
   endif()
-  _ament_cmake_gtest_find_gtest()
+  find_package(GTest REQUIRED)
 endmacro()
