diff --git a/pluginlib/CMakeLists.txt b/pluginlib/CMakeLists.txt
index cd0749f..9af3e07 100644
--- a/pluginlib/CMakeLists.txt
+++ b/pluginlib/CMakeLists.txt
@@ -15,15 +15,15 @@ find_package(ament_index_cpp REQUIRED)
 find_package(class_loader REQUIRED)
 find_package(rcutils REQUIRED)
 find_package(rcpputils REQUIRED)
-find_package(tinyxml2_vendor REQUIRED)
-find_package(TinyXML2 REQUIRED)  # provided by tinyxml2 upstream, or tinyxml2_vendor
+find_package(tinyxml2 REQUIRED)
 
 add_library(${PROJECT_NAME} INTERFACE)
 target_include_directories(${PROJECT_NAME} INTERFACE
   "$<INSTALL_INTERFACE:include>")
+target_link_libraries(${PROJECT_NAME} INTERFACE tinyxml2::tinyxml2)
 ament_target_dependencies(${PROJECT_NAME} INTERFACE
-  ament_index_cpp class_loader rcutils rcpputils TinyXML2)
-ament_export_dependencies(ament_index_cpp class_loader rcutils rcpputils tinyxml2_vendor TinyXML2)
+  ament_index_cpp class_loader rcutils rcpputils)
+ament_export_dependencies(ament_index_cpp class_loader rcutils rcpputils tinyxml2)
 ament_export_include_directories(include)
 ament_export_targets(${PROJECT_NAME})
 
@@ -116,13 +116,14 @@ if(BUILD_TESTING)
     APPEND_ENV AMENT_PREFIX_PATH=${mock_install_path}
   )
   if(TARGET ${PROJECT_NAME}_unique_ptr_test)
+    target_link_libraries(${PROJECT_NAME}_unique_ptr_test tinyxml2::tinyxml2)
     ament_target_dependencies(
       ${PROJECT_NAME}_unique_ptr_test
       class_loader
       ament_index_cpp
       rcpputils
       rcutils
-      TinyXML2
+      tinyxml2
     )
     target_compile_definitions(${PROJECT_NAME}_unique_ptr_test PUBLIC "PLUGINLIB__DISABLE_BOOST_FUNCTIONS")
 
@@ -137,13 +138,14 @@ if(BUILD_TESTING)
     APPEND_ENV AMENT_PREFIX_PATH=${mock_install_path}
   )
   if(TARGET ${PROJECT_NAME}_utest)
+    target_link_libraries(${PROJECT_NAME}_utest tinyxml2::tinyxml2)
     ament_target_dependencies(
       ${PROJECT_NAME}_utest
       class_loader
       ament_index_cpp
       rcpputils
       rcutils
-      TinyXML2
+      tinyxml2
     )
     target_compile_definitions(${PROJECT_NAME}_utest PUBLIC "PLUGINLIB__DISABLE_BOOST_FUNCTIONS")
 
diff --git a/pluginlib/package.xml b/pluginlib/package.xml
index 0b2b69c..d136f6f 100644
--- a/pluginlib/package.xml
+++ b/pluginlib/package.xml
@@ -25,7 +25,7 @@
   <depend>class_loader</depend>
   <depend>rcutils</depend>
   <depend>rcpputils</depend>
-  <depend>tinyxml2_vendor</depend>
+  <depend>tinyxml2</depend>
 
   <test_depend>ament_cmake_gtest</test_depend>
 
