diff --git a/resource_retriever/CMakeLists.txt b/resource_retriever/CMakeLists.txt
index 5198148..3ecc476 100644
--- a/resource_retriever/CMakeLists.txt
+++ b/resource_retriever/CMakeLists.txt
@@ -13,7 +13,7 @@ endif()
 find_package(ament_cmake_ros REQUIRED)
 find_package(ament_index_cpp REQUIRED)
 # TODO(wjwwood): remove libcurl_vendor and just use system curl when possible
-find_package(libcurl_vendor REQUIRED)
+find_package(CURL REQUIRED)
 
 # TODO(wjwwood): split cpp and python apis into separate packages
 
@@ -23,9 +23,9 @@ target_include_directories(${PROJECT_NAME}
     $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
     $<INSTALL_INTERFACE:include>
 )
+target_link_libraries(${PROJECT_NAME} CURL::libcurl)
 ament_target_dependencies(${PROJECT_NAME}
   ament_index_cpp
-  libcurl_vendor
 )
 # Causes the visibility macros to use dllexport rather than dllimport,
 # which is appropriate when building the dll but not consuming it.
@@ -35,8 +35,7 @@ target_compile_definitions(${PROJECT_NAME} PRIVATE "RESOURCE_RETRIEVER_BUILDING_
 ament_export_include_directories(include)
 ament_export_targets(${PROJECT_NAME} HAS_LIBRARY_TARGET)
 
-ament_export_dependencies(ament_index_cpp)
-ament_export_dependencies(libcurl_vendor)
+ament_export_dependencies(ament_index_cpp CURL)
 
 ament_python_install_package(${PROJECT_NAME}
   PACKAGE_DIR src/${PROJECT_NAME})
diff --git a/resource_retriever/package.xml b/resource_retriever/package.xml
index a656b39..512bfc1 100644
--- a/resource_retriever/package.xml
+++ b/resource_retriever/package.xml
@@ -25,7 +25,6 @@
 
   <depend>ament_index_cpp</depend>
   <depend>ament_index_python</depend>
-  <depend>libcurl_vendor</depend>
 
   <test_depend>ament_cmake_gtest</test_depend>
   <test_depend>ament_cmake_pytest</test_depend>
