diff --git a/rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake b/rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake
index 18f8378..a466ea8 100644
--- a/rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake
+++ b/rosidl_generator_py/cmake/rosidl_generator_py_generate_interfaces.cmake
@@ -185,6 +185,11 @@ target_link_libraries(
   ${_target_name_lib}
   ${PythonExtra_LIBRARIES}
 )
+
+if(QNX)
+  list(APPEND PythonExtra_INCLUDE_DIRS ${NUMPY_HEADERS})
+endif()
+
 target_include_directories(${_target_name_lib}
   PUBLIC
   ${CMAKE_CURRENT_BINARY_DIR}/rosidl_generator_c
